pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,187 @@
+		return fmt.Errorf("error reconciling fleet workspace verbs: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(globalRoleBinding *v3.GlobalRoleBinding, globalRole *v3.GlobalRole, fleetWorkspaces []*v3.FleetWorkspace) error {
+	var returnError error
+	subject := rbac.GetGRBSubject(globalRoleBinding)
+	roleref := v1.RoleRef{
+		APIGroup: "rbac.authorization.k8s.io",
+		Kind:     "ClusterRole",
+		Name:     wrangler.SafeConcatName(globalRole.Name, fleetWorkspaceClusterRulesName),
+	}
+	subjects := []v1.Subject{
+		subject,
+	}
+
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+
+		rbName := wrangler.SafeConcatName(globalRoleBinding.Name)
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, rbName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			returnError = multierror.Append(returnError, err)
+			continue
+		}
+		if apierrors.IsNotFound(err) {
+			_, err = h.rbClient.Create(
+				&v1.RoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      rbName,
+						Namespace: fleetWorkspace.Name,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: v3.GlobalRoleBindingGroupVersionKind.GroupVersion().String(),
+								Kind:       v3.GlobalRoleBindingGroupVersionKind.Kind,
+								Name:       globalRoleBinding.Name,
+								UID:        globalRoleBinding.UID,
+							},
+						},
+						Labels: map[string]string{
+							grbOwnerLabel:                 wrangler.SafeConcatName(globalRoleBinding.Name),
+							fleetWorkspacePermissionLabel: "true",
+							controllers.K8sManagedByKey:   controllers.ManagerValue,
+						},
+					},
+					RoleRef:  roleref,
+					Subjects: subjects,
+				})
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+			}
+		} else if !reflect.DeepEqual(rb.Subjects, subjects) ||
+			reflect.DeepEqual(rb.RoleRef, roleref) {
+			// undo modifications if rb has changed.
+			rb.Subjects = subjects
+			rb.RoleRef = roleref
+			_, err := h.rbClient.Update(rb)
+			if err != nil {
+				returnError = multierror.Append(returnError, err)
+			}
+		}
+	}
+
+	return returnError
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileWorkspaceVerbsBindings(globalRoleBinding *v3.GlobalRoleBinding, globalRole *apimgmtv3.GlobalRole, fleetWorkspaces []*apimgmtv3.FleetWorkspace) error {
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil || len(globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs) == 0 {
+		return nil
+	}
+	crbName := wrangler.SafeConcatName(globalRoleBinding.Name, fleetWorkspaceVerbsName)
+	crb, err := h.crbCache.Get(crbName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	subjects := []v1.Subject{rbac.GetGRBSubject(globalRoleBinding)}
+	roleref := v1.RoleRef{
+		APIGroup: "rbac.authorization.k8s.io",
```

Comment:
Nit:
Here and elsewhere, you could use the `rbacv1.GroupName` constant from the import `rbacv1 "k8s.io/api/rbac/v1"`

Reply:
added in the latest commit https://github.com/rancher/rancher/pull/44901/commits/723d79e964a44843ce33dce0e3264ece4ac7a5c8